XenModel/PoolJoinRules.cs

```diff
@@ -382,6 +388,27 @@ private static bool DifferentServerVersion(Host slave, Host master)
                 slave.ProductBrand != master.ProductBrand;
         }
 
+        private static bool DifferentPatches(Host slave, Host master)
+        {
+            if (Helpers.ElyOrGreater(master))
+            {
+                var masterUpdates = master.AppliedUpdates();
+                var slaveUpdates = slave.AppliedUpdates();
+                if (masterUpdates.Except(slaveUpdates, new UpdateUuidEqualityComparer()).Any() ||

```

Comment:
I'm not sure why we're defining new classes UpdateUuidEqualityComparer and PatchUuidEqualityComparer. Can't we just make the default Equals do the right thing on updates and patches. (Or maybe it does the right thing already?)